single_kernel_mongo/utils/mongodb_users.py

```diff
@@ -190,3 +198,30 @@ def get_user_from_username(username: str) -> MongoDBUser:
     if username == LogRotateUser.username:
         return LogRotateUser
     raise ValueError(f"Unknown user: {username}")
+
+
+def validate_charm_user_password_config(user_passwords: dict):
+    """Validate a mapping of charm usernames to passwords.
+
+    Rules:
+      - Only internal charmed usernames are allowed.
+      - Passwords must not be empty or only whitespace.
+      - Passwords must not exceed MAX_PASSWORD_LENGTH characters.
+
+    Raises:
+        InvalidPasswordError: if any validation rule is violated.
+    """
+    extra_users = set(user_passwords.keys()) - CharmUsernames
+    if extra_users:
+        raise InvalidPasswordError(f"Unexpected usernames provided: {', '.join(extra_users)}")
```

Comment:
I'm not sure of the specification but my understanding was that we should support a secret with extra data that we just ignore, not raise if there's any unexpected field.
@reneradoi Could you confirm/infirm that assertion?